src/visualizers/VisualizerGrid.ts

```diff
@@ -0,0 +1,1040 @@
+            color: RED,
+            aux: 3n,
+        },
+        {
+            property: Property.Polygonal_Number,
+            visualization: PropertyVisualization.Fill_Cell,
+            color: ORANGE,
+            aux: 4n,
+        },
+        {
+            property: Property.Polygonal_Number,
+            visualization: PropertyVisualization.Fill_Cell,
+            color: YELLOW,
+            aux: 5n,
+        },
+        {
+            property: Property.Polygonal_Number,
+            visualization: PropertyVisualization.Fill_Cell,
+            color: GREEN,
+            aux: 6n,
+        },
+        {
+            property: Property.Polygonal_Number,
+            visualization: PropertyVisualization.Fill_Cell,
+            color: BLUE,
+            aux: 7n,
+        },
+        {
+            property: Property.Polygonal_Number,
+            visualization: PropertyVisualization.Fill_Cell,
+            color: PURPLE,
+            aux: 8n,
+        },
+    ],
+    Color_By_Last_Digit_1: Array.from({length: 10}, (_item, index) => ({
+        property: Property.Last_Digit_Is,
+        visualization: PropertyVisualization.Fill_Cell,
+        color: RAINBOW[index],
+        aux: BigInt(index),
+    })),
+    Color_By_Last_Digit_2: Array.from({length: 10}, (_item, index) => ({
+        property: Property.Last_Digit_Is,
+        visualization:
+            index % 2
+                ? PropertyVisualization.Box_In_Cell
+                : PropertyVisualization.Fill_Cell,
+        color: RAINBOW[index],
+        aux: BigInt(index),
+    })),
+}
+
+function getPropertyParams(index: number, prop: PropertyObject) {
+    return {
+        [`property${index}`]: {
+            value: prop.property,
+            from: Property,
+            displayName: `Property ${index + 1}`,
+            required: false,
+            visibleDependency: index > 0 ? `property${index - 1}` : '',
+            visiblePredicate: (d: Property) => d !== Property.None,
+        },
+        [`prop${index}Vis`]: {
+            value: prop.visualization,
+            from: PropertyVisualization,
+            displayName: 'Display',
+            required: false,
+            visibleDependency: `property${index}`,
+            visiblePredicate: (d: Property) => d !== Property.None,
+        },
+        [`prop${index}Color`]: {
+            value: prop.color,
+            forceType: 'color',
+            displayName: 'Color',
+            required: false,
+            visibleDependency: `property${index}`,
+            visiblePredicate: (d: Property) => d !== Property.None,
+        },
+        [`prop${index}Aux`]: {
+            value: prop.aux,
+            displayName: (d: Property) => propertyAuxName[Property[d]] || '',
+            required: false,
+            visibleDependency: `property${index}`,
+            visiblePredicate: (d: Property) => Property[d] in propertyAuxName,
+        },
+    }
+}
+
+/*
+ *   FUNCTIONS TO CHECK FOR PROPERTIES
```

Comment:
I will add a commit that strips the unnecessary leading white space from this comment and adds more information about the property functions.

Reply:
Done in https://github.com/numberscope/frontscope/pull/230/commits/9cddb1cbd00ffae47823820f576cc37b78e76f91.